rules/azurerm_linux_virtual_machine_invalid_admin_username.go

```diff
@@ -0,0 +1,109 @@
+package rules
+
+// Please see: https://docs.microsoft.com/en-us/azure/virtual-machines/linux/faq#what-are-the-username-requirements-when-creating-a-vm
+
+import (
+	"fmt"
+	"strings"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-azurerm/project"
+)
+
+// AzurermLinuxVirtualMachineInvalidAdminUserNameRule ensures that invalid usernames are not set.
+type AzurermLinuxVirtualMachineInvalidAdminUserNameRule struct {
+	resourceType  string
+	attributeName string
+	enum          []string
+}
+
+// NewAzurermLinuxVirtualMachineInvalidAdminUserNameRule returns new rule with default attributes
+func NewAzurermLinuxVirtualMachineInvalidAdminUserNameRule() *AzurermLinuxVirtualMachineInvalidAdminUserNameRule {
+	return &AzurermLinuxVirtualMachineInvalidAdminUserNameRule{
+		resourceType:  "azurerm_linux_virtual_machine",
+		attributeName: "admin_username",
+		enum: []string{
+			"azureuser",
```

Comment:
Also, it is too hard to maintain the invalid username list in each rule, so it may be a good idea to cut it out as a function in another file :)
Here is an example: https://github.com/terraform-linters/tflint-ruleset-google/blob/54c31d26684babcb09d3b3ca3b22304af58effc9/rules/utils.go

Reply:
Definitely a good idea. I refactored the code so that the invalid usernames come from a shared utility object. In terms of AzureUser. Yes, it's not technically on the list, but our team had some discussions and we feel it's a best practice to avoid it if possible. 

This is because AzureUser is the default name assigned in the UI and is becoming similar to "root" in that people can easily guess it. A best practice from a security point of view is to avoid these guessable names and that includes defaults.